### DISCUSSION TITLE
How can i set the state of an atom outside of react?

### DISCUSSION THREAD

**Discussion by ivan-angjelkoski** [pmndrs/jotai#1875] [Q&A]
How can i set the state of an atom outside of react?
 suppose we have the following atom: 
 ```
const counterAtom = atom(1)
```

how can we change the state without using the `useAtom` hook ?
like for example is there a method like this? :
```
counterAtom.setState(state => state+ 1 )
```

**Comment by arjunvegda** [comment]
Jotai 2.0 exposes setters and getters on the store method. Here is an example:

```ts
import { atom, getDefaultStore } from 'jotai';

const defaultStore = getDefaultStore();

const counterAtom = atom(1)

defaultStore.set(counterAtom, (state) => state + 1)
```

Docs: https://jotai.org/docs/core/store
